admin-dev/themes/new-theme/js/pages/product/components/catalog-price-rule/catalog-price-rule-renderer.ts

```diff
@@ -0,0 +1,145 @@
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+import {EventEmitter} from 'events';
+import ProductMap from '@pages/product/product-map';
+import RendererType from '@PSTypes/renderers';
+
+const CatalogPriceRuleMap = ProductMap.catalogPriceRule;
+
+export default class CatalogPriceRuleRenderer implements RendererType {
+  private eventEmitter: EventEmitter;
+
+  private listContainer: HTMLElement
+
+  private $loadingSpinner: JQuery;
+
+  private $listTable: JQuery;
+
+  constructor() {
+    this.listContainer = <HTMLElement>document.querySelector(CatalogPriceRuleMap.listContainer);
+    this.eventEmitter = window.prestashop.instance.eventEmitter;
+    this.$loadingSpinner = $(ProductMap.catalogPriceRule.loadingSpinner);
+    this.$listTable = $(ProductMap.catalogPriceRule.listTable);
+  }
+
+  public setLoading(loading: boolean): void {
+    this.$loadingSpinner.toggle(loading);
+    this.$listTable.toggle(!loading);
+  }
+
+  public render(data: Record<string, any>): void {
+    console.log(data);
+    const {listFields} = CatalogPriceRuleMap;
+    const tbody = this.listContainer.querySelector<HTMLElement>(`${CatalogPriceRuleMap.listContainer} tbody`);
+
+    if (!tbody) {
+      console.log(`Error: ${CatalogPriceRuleMap.listContainer} element not found`);
+      return;
+    }
+    const trTemplateContainer = this.listContainer.querySelector<HTMLScriptElement>(CatalogPriceRuleMap.listRowTemplate);
+
+    if (!trTemplateContainer) {
+      console.log(`Error: ${CatalogPriceRuleMap.listRowTemplate} element not found`);
+      return;
+    }
+    const rowContainer = document.querySelector<HTMLElement>(CatalogPriceRuleMap.blockContainer);
+
+    if (!rowContainer) {
+      console.log(`Error: ${CatalogPriceRuleMap.blockContainer} element not found`);
+      return;
+    }
+    const editCatalogPriceRuleUrl = rowContainer.dataset.catalogPriceUrl;
+
+    if (!editCatalogPriceRuleUrl) {
+      console.log('Error: Catalog price rule url not found');
```

Comment:
```suggestion
      console.error('Error: Catalog price rule url not found');
```
same goes for all logs above